gateway/policy.go

```diff
@@ -186,14 +186,18 @@ func parsePoliciesFromRPC(list string) (map[string]user.Policy, error) {
 	policies := make(map[string]user.Policy, len(dbPolicyList))
 
 	for _, p := range dbPolicyList {
-		p.ID = p.MID.Hex()
-		policies[p.MID.Hex()] = p
+		id := p.MID.Hex()
+		if allowExplicit && p.ID != "" {
+			id = p.ID
+		}
+		p.ID = id
```

Comment:
can we change this block to something like:

```
if !(allowExplicit && pid != "") {
		pid = id
	}
```

We will only modify p.ID when it mets that condition. And do not assign everytime? because under the hood sometimes we are assigning the same p.ID to p.ID

Reply:
@sredxny `p.ID` is not always populated. By default, `p.MID` is used to override `p.ID`.